actions/restore.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+from charms import layer
+from charmhelpers.core.hookenv import action_fail
+from charmhelpers.core.hookenv import action_get
+from charmhelpers.core.hookenv import action_set
+from charmhelpers.core.hookenv import config
+from charmhelpers.core.hookenv import log
+from charmhelpers.core.hookenv import unit_get
+from charmhelpers.core.hookenv import resource_get
+from charmhelpers.core.host import chdir
+from charmhelpers.core.host import service_start
+from charmhelpers.core.host import service_stop
+from shlex import split
+from subprocess import check_call
+from subprocess import check_output
+from subprocess import Popen
+from subprocess import PIPE
+from datetime import datetime
+import hashlib
+import os
+import sys
+import time
+
+opts = layer.options('etcd')
+
+DATESTAMP = datetime.strftime(datetime.now(), '%Y%m%d-%H%M%S')
+ARCHIVE = "etcd-data-{}.tar.gz".format(DATESTAMP)
+
+ETCD_DATA_DIR = opts['etcd_data_dir']
+ETCD_PORT = config('management_port')
+PRIVATE_ADDRESS = unit_get('private-address')
+SKIP_BACKUP = action_get('skip-backup')
+SNAPSHOT_ARCHIVE = resource_get('snapshot')
+TARGET_PATH = action_get('target')
+
+
+def preflight_check():
+    ''' Check preconditions for data restoration '''
+    if not SNAPSHOT_ARCHIVE:
+        action_fail({'result.failed': 'Missing snapshot. See: README.md'})
+        sys.exit(0)
+
+
+def render_backup():
+    ''' Backup existing data in the event of restoration on a dirty unit. '''
+    if not os.path.isdir(ETCD_DATA_DIR):
```

Comment:
Since we're setting SKIP_BACKUP to default true, should we log if there's no data dir and SKIP_BACKUP is false?